### PR TITLE
Add wg-k8s-infra with draft charter

### PR DESCRIPTION
**Pull request by spiffxp**
There are some TODO's around renaming external assets from
k8s-infra-team to wg-k8s-infra, but I think that can be done
in a follow up PR. My main concern here is whether the
charter looks good enough to move forward

Listing as WIP while I see how this aligns with the current corpus of WG
docs from steering.

Fixes https://github.com/kubernetes/community/issues/2715

Content pulled from: https://docs.google.com/document/d/1zLh-YkLN3P09yyQf534VJCmiYUcqezh7HRHKyf7CenQ/edit

### CODE REVIEW

File: wg-k8s-infra/charter.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+### Out of Scope
+
+- We are not responsible for maintaining infrastructure which has not yet been
+  migrated to the CNCF. For example, problems with prow.k8s.io should be routed
+  to SIG Testing
+
+## Roles and Organization Management
+
+This sig adheres to the Roles and Organization Management outlined in
```
Comment (thockin) [inline wg-k8s-infra/charter.md]: s/sig/wg

**Comment by spiffxp** [comment]
/hold
mostly testing whether k8s-ci-robot is responding now

**Comment by spiffxp** [comment]
I removed the WIP, and think this is ready for review.

File: sig-list.md [hunk 2]
```diff
@@ -62,6 +62,7 @@ When the need arises, a [new SIG can be created](sig-creation-procedure.md)
...
 |[IoT Edge](wg-iot-edge/README.md)|* [Dejan Bosanac](https://github.com/dejanb), Red Hat<br>|* [Slack](https://kubernetes.slack.com/messages/wg-iot-edge)<br>* [Mailing List](https://groups.google.com/forum/#!forum/kubernetes-wg-iot-edge)|* Regular WG Meeting: [Fridays at 15:00 UTC (bi-weekly)](https://docs.google.com/document/d/1FQx0BPlkkl1Bn0c9ocVBxYIKojpmrS1CFP5h0DI68AE/edit)<br>
+|[K8s Infra](wg-k8s-infra/README.md)|* [Davanum Srinivas](https://github.com/dims), Huawei<br>* [Aaron Crickenberger](https://github.com/spiffxp), Google<br>|* [Slack](https://kubernetes.slack.com/messages/k8s-infra-team)<br>* [Mailing List](https://groups.google.com/forum/#!forum/k8s-infra-team)|* Regular WG Meeting: [Fridays at 9:00 PT (Pacific Time) (bi-weekly)](https://docs.google.com/document/d/1FQx0BPlkkl1Bn0c9ocVBxYIKojpmrS1CFP5h0DI68AE/edit)<br>
```
Comment (dims) [inline sig-list.md]: wrong url @spiffxp ?

File: sigs.yaml [hunk 3]
```diff
@@ -2144,3 +2144,30 @@ workinggroups:
...
+      - name: Aaron Crickenberger
+        github: spiffxp
+        company: Google
+    meetings:
+    - description: Regular WG Meeting
+      day: Friday
+      time: "9:00"
+      tz: "PT (Pacific Time)"
+      frequency: bi-weekly
+      url: https://docs.google.com/document/d/1FQx0BPlkkl1Bn0c9ocVBxYIKojpmrS1CFP5h0DI68AE/edit
```
Comment (dims) [inline sigs.yaml]: Same here

File: sig-list.md (hunk 2 above)
Comment (spiffxp) [reply sig-list.md]: This is the example I've seen in other places, the meeting time used to be a zoom link, and now it's a link to this doc explaining why we can't publicly post zoom links

Comment (dims) [reply sig-list.md]: Ah. Sorry! got it now.

**Comment by dims** [comment]
LGTM 👍

File: wg-k8s-infra/charter.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+External facing services implemented as subprojects. They often span multiple
+SIGs in terms of ownership, hence why this WG is acting as a steward of their
+migration.
+
+| Service | SIG(s) | Notes |
+| --- | --- | --- |
+| DNS | Contribex, Release | Domain Name Services for Kubernetes assets |
+| GCR | Release | Repository for Container Images |
+| GCS / Object Storage | Release, Testing | Buckets for logs, test arifacts, APT, RPM |
+| GKE + Stack Driver| Contribex, Release, Testing | Clusters for running bots, utilities, prow, etc |
```
Comment (ixdy) [inline wg-k8s-infra/charter.md]: we also use GAE for gubernator and testgrid.

File: wg-k8s-infra/charter.md [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+
+External facing services implemented as subprojects. They often span multiple
+SIGs in terms of ownership, hence why this WG is acting as a steward of their
+migration.
+
+| Service | SIG(s) | Notes |
+| --- | --- | --- |
+| DNS | Contribex, Release | Domain Name Services for Kubernetes assets |
+| GCR | Release | Repository for Container Images |
+| GCS / Object Storage | Release, Testing | Buckets for logs, test arifacts, APT, RPM |
```
Comment (ixdy) [inline wg-k8s-infra/charter.md]: typo: artifacts

we also use GCS for release tarballs.

File: wg-k8s-infra/charter.md [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+#### Code, Binaries and Services
+
+External facing services implemented as subprojects. They often span multiple
+SIGs in terms of ownership, hence why this WG is acting as a steward of their
+migration.
+
+| Service | SIG(s) | Notes |
+| --- | --- | --- |
+| DNS | Contribex, Release | Domain Name Services for Kubernetes assets |
+| GCR | Release | Repository for Container Images |
```
Comment (ixdy) [inline wg-k8s-infra/charter.md]: should we also mention GCB (google cloud builder), which is currently used to run releases?

File: wg-k8s-infra/charter.md [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+- **High risk changes** likely break existing contributor workflows, may be
+  difficult to roll back, and likely impact all of the project's repos. These
+  require a consultation with SIG Contributor Experience as well as any other
+  owning SIGs, and a lazy consensus issue with [kubernetes-dev@] notice.
+
+### Out of Scope
+
+- We are not responsible for maintaining infrastructure which has not yet been
+  migrated to the CNCF. For example, problems with prow.k8s.io should be routed
+  to SIG Testing
```
Comment (ixdy) [inline wg-k8s-infra/charter.md]: nit: missing period

File: wg-k8s-infra/charter.md (hunk 4 above)
Comment (spiffxp) [reply wg-k8s-infra/charter.md]: Added.  Maybe it's too optimistic to hope we'd have GKE-hosted variants of these by the time we need to migrate them.

File: wg-k8s-infra/charter.md (hunk 6 above)
Comment (spiffxp) [reply wg-k8s-infra/charter.md]: Added

**Comment by bgrant0607** [comment]
@spiffxp Is this a time-bounded effort?

File: sigs.yaml [hunk 8]
```diff
@@ -2144,3 +2144,30 @@ workinggroups:
       tz: "PT (Pacific Time)"
       frequency: weekly
       url: https://docs.google.com/document/d/1RbC4SBZBlKth7IjYv_NaEpnmLGwMJ0ElpUOmsG-bdRA/edit
+  - name: K8s Infra
+    dir: wg-k8s-infra
+    mission_statement: >
+      A Working Group dedicated to migrating Kubernetes project infrastructure
+      over to the CNCF, and the creation of teams and processes for ongoing
+      maintenance. Involves collaboration with multiple SIGs such as
+      Architecture, Contributor Experience, Release, and Testing.
```
Comment (timothysc) [inline sigs.yaml]: Testing, etc.

File: sigs.yaml [hunk 9]
```diff
@@ -2144,3 +2144,30 @@ workinggroups:
...
+      - name: Davanum Srinivas
+        github: dims
+        company: Huawei
+      - name: Aaron Crickenberger
+        github: spiffxp
+        company: Google
+    meetings:
+    - description: Regular WG Meeting
+      day: Friday
+      time: "9:00"
```
Comment (timothysc) [inline sigs.yaml]: Is this set?  @dims was mentioning trying to find another time.

File: sigs.yaml (hunk 8 above)
Comment (spiffxp) [reply sigs.yaml]: Is there any etc here? I'd like to eventually have working groups list out the sigs they're tied to, so if this list is incomplete I'd like to fill it out.  I think there will be downstream consumer SIGs like cluster-lifecycle, I tried to keep this list to sigs that seem to canonically "own" these infra today.

File: sigs.yaml (hunk 9 above)
Comment (spiffxp) [reply sigs.yaml]: I'm happy to change this post-Doodle poll, either in this PR or a followup depending on when we resolve

**Comment by spiffxp** [comment]
Discussed with steering, objection was raised that this reads "too sig like" and may further muddy the waters of what WG's are and are not vs. SIG's, will work with @dims to wordsmith.

**Comment by jdumars** [comment]
/approve

someone else please feel free to remove the hold when appropriate

**Comment by BenTheElder** [comment]
needs rebase, probing for signs of life 🛸

**Comment by dims** [comment]
:)

**Comment by spiffxp** [comment]
Charter's been rebased, I squashed the previous commits.  Now looking at changes necessary to address steering concerns

File: sigs.yaml (hunk 8 above)
Comment (spiffxp) [reply sigs.yaml]: Have added the etc, will want to revisit when we quantify list of SIGs (ref: https://github.com/kubernetes/community/issues/2176)

**Comment by dims** [comment]
LGTM @spiffxp

**Comment by spiffxp** [comment]
Sounds like this is ready for a proper `/lgtm`, I'll remove the `/hold` when I get one

**Comment by philips** [comment]
LGTM

**Comment by dims** [comment]
LGTM

**Comment by spiffxp** [comment]
Context: https://groups.google.com/a/kubernetes.io/d/msg/steering/FC7bblZLZ_I/HxK1zo-OCwAJ

I forgot to get written LGTMs from steering committee and asked for them retroactively.  There are now LGTMs from a simple majority of the steering committee.

**Pull request merged**